### DISCUSSION TITLE
Disable include path completion suggestion insertion on entering a path separator

### DISCUSSION THREAD

**Discussion by Neonit** [microsoft/vscode-cpptools#7055] [Q&A] [answered]
Hello,

(**TL;DR**) How can I disable include path completion insertion on typing a path separator character?

Background:
When typing an include like (like `#include <foo`, where *foo* is the exact and complete folder name I want to use) a completion suggestion list pops up. Unfortunately when I enter a path separator (`/` or `\`) it automatically chooses the first entry and inserts it. Due to fuzzy matching it often (way too often) replaces the previously typed path segment with something completely different (to continue the example: `#include <forever_oppressing_your_workflow.h/`, because *foo* got fuzzy-matched with *forever_oppressing_your_workflow*). Even with leaves (header files), which makes no sense, because why would I enter a path separator, if the previous segment is a leaf?

Of course the reason for it not completing to *foo* is that it doesn't have *foo* in its include path at the time, but I often run into situations where this is (temporarily) the case. Probably because I'm using CMake and the CMake extension wants me to reconfigure for each new file I create and I usually procrastinate that to a later point in time.

All I want is to disable this feature (inserting completion suggestion on path sep). I have my <kdb>Tab</kbd> key for inserting completion suggestions and I'm fine with that. I have searched the settings and the web, but couldn't find a way to disable it. I hope I just oversaw it and you can point me to the right setting.

**Comment by sean-mcmanus** [comment, marked as answer]
There is no setting for that. I could transfer this to a feature request.

**Reply by Neonit** [reply]
Please do that. Thank you.

**Reply by sean-mcmanus** [reply]
https://github.com/microsoft/vscode-cpptools/issues/7084
